src/toolkits/drawing_classifier/dc_data_iterator.hpp

```diff
@@ -0,0 +1,178 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef TURI_DRAWING_CLASSIFICATION_DC_DATA_ITERATOR_HPP_
+#define TURI_DRAWING_CLASSIFICATION_DC_DATA_ITERATOR_HPP_
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include <core/data/sframe/gl_sframe.hpp>
+#include <ml/neural_net/float_array.hpp>
+
+
+namespace turi {
+namespace drawing_classifier {
+
+constexpr int kDrawingHeight = 28;
+constexpr int kDrawingWidth = 28;
+constexpr int kDrawingChannels = 1;
+
+/**
+ * Pure virtual interface for classes that produce batches of data
+ * (pre-augmentation) from a raw SFrame.
+ */
+class data_iterator {
```

Comment:
Could we add a TODO to factor out the shared structure with the other iterators?

Reply:
Done!